packages/big-design/src/components/ButtonGroup/styled.tsx

```diff
@@ -0,0 +1,57 @@
+import { theme as defaultTheme } from '@bigcommerce/big-design-theme';
+import styled, { css } from 'styled-components';
+
+import { StyleableButton } from '../Button/private';
+import { FlexItem } from '../Flex';
+
+interface StyledButtonProps {
+  borderRadius?: boolean;
+}
+
+interface StyledFlexItemProps {
+  isVisible: boolean;
+}
+
+export const StyledButton = styled(StyleableButton)<StyledButtonProps>`
+  ${({ borderRadius, theme }) =>
+    borderRadius
+      ? css`
+          border-radius: ${theme.borderRadius.normal};
+        `
+      : css`
+          border-radius: ${theme.borderRadius.none};
+        `}
+
+  &:focus {
+    z-index: 1;
+  }
+`;
+
+export const StyledFlexItem = styled(FlexItem)<StyledFlexItemProps>`
+  margin-right: -1px;
```

Comment:
Here I'm resolving a problem with the double border. I use pixels cause we have a `1px` border in the theme package.

Reply:
I'd rather not use negative margins here, could we use something like this instead or does that cause an issue?
```tsx
  &:not(:first-of-type) ${StyledButton} {
    border-left: 0;
  }
```

Reply:
Good catch! I tried to follow this way but, yeah, with this case I get the problem when buttons jumping horizontally on focus cause I need to restore the border on the focus state that buttons will look normal.

So, I think I could use `border-left: 0` with `border-color: transparent` but, we have a transition for `border-color`...  I research the same case around other libs and looks like negative margin is normal 🤷

Reply:
Ahh, good point. I'm good with this approach then 👌